Remove django 20 warnings

umgelurgel opened the pull request:
Branch based off https://github.com/EuroPython/epcon/pull/847 to avoid merge conflicts. 847 should be reviewed/merged first.

before:
```
-- Docs: https://docs.pytest.org/en/latest/warnings.html
============ 129 passed, 35 skipped, 1573 warnings in 29.28 seconds ============
```

after:
```
-- Docs: https://docs.pytest.org/en/latest/warnings.html
============ 129 passed, 35 skipped, 1332 warnings in 27.70 seconds ============
```

Most remaining warnings are generated either by django-cms or are pytest deprecation warnings.

artcz commented:
This one needs rebase. :)

umgelurgel commented:
Rebased, kind of - it was easier to squash the commits then create a new branch off the already rebased `make-comments-flat` :)

artcz commented on assopy/models.py:
```diff
@@ -349,7 +349,7 @@ class Coupon(models.Model):
     value = models.CharField(max_length=8, help_text='importo, eg: 10, 15%, 8.5')
 
-    user = models.ForeignKey(User, null=True, blank=True)
+    user = models.ForeignKey(User, null=True, blank=True, on_delete=models.CASCADE)
```
is that using the same 'User' as `get_user_model()` in some of the changes above?

umgelurgel replied:
No, this is actually the `assopy.User` defined earlier in the file. Renaming that model to AssopyUser was definitely a good idea.

artcz commented on assopy/models.py:
```diff
@@ -582,8 +582,8 @@ def __str__(self):
 class VatFare(models.Model):
-    fare = models.ForeignKey('conference.fare')
-    vat = models.ForeignKey(Vat)
+    fare = models.ForeignKey('conference.fare', on_delete=models.CASCADE)
```
unless this causes some big circular import I'd opt for using class here directly instead of the string.

artcz commented on assopy/models.py:
```diff
@@ -582,8 +582,8 @@ def __str__(self):
-    fare = models.ForeignKey('conference.fare')
-    vat = models.ForeignKey(Vat)
+    fare = models.ForeignKey('conference.fare', on_delete=models.CASCADE)
+    vat = models.ForeignKey(Vat, on_delete=models.CASCADE)
```
BTW, in general I'm more of a fan of SET_NULL or DO_NOTHING than CASCADE, but because cascade is default I guess it would require reviewing every single FK – so it's probably better to move them to explicit  CASCADE (for django20 warnings), like in this PR, and then maybe update them to some other values where it makes sense.

artcz commented on assopy/models.py:
```diff
@@ -976,8 +976,8 @@ def net_price(self):
 class RefundOrderItem(models.Model):
-    orderitem = models.ForeignKey('assopy.OrderItem')
-    refund = models.ForeignKey('assopy.Refund')
+    orderitem = models.ForeignKey('assopy.OrderItem', on_delete=models.CASCADE)
```
same here, would replace those with classes (ie `OrderItem`) unless it causes circular. If it does – maybe add a NOTE instead.

artcz commented on common/jsonify.py:
```diff
@@ -13,9 +15,10 @@ def default(self, obj):
             # required to support User.is_authenticated from Django 1.10 onwards
-            return obj()
+            # and avoid depracation warnings
+            return obj == True
```
clever. :+1:

artcz commented on conference/forms.py:
```diff
@@ -74,7 +74,7 @@ def _media(self):
     media = property(_media)
 
-    def render(self, name, value, attrs=None):
+    def render(self, name, value, attrs=None, renderer=None):
```
i assume it's for some inheritance/API compatibility, that said, renderer is passed as argument but not used in the function.

umgelurgel replied:
Exactly, it's for compatibility with the new signature of the `render` method.

artcz commented on conference/forms.py:
```diff
@@ -96,7 +96,7 @@ def render(self, name, value, attrs=None):
 
 class ReadonlyTagWidget(widgets.TextInput):
-    def render(self, name, value, attrs=None):
+    def render(self, name, value, attrs=None, renderer=None):
```
ditto, renderer

artcz commented on conference/forms.py:
```diff
@@ -145,7 +145,7 @@ class PseudoRadioSelectWidget(forms.widgets.RadioSelect):
 
     def get_context(self, name, value, attrs):
-        context = super(PseudoRadioSelect, self).get_context(name, value, attrs)
+        context = super(PseudoRadioSelectWidget, self).get_context(name, value, attrs)
```
we're python3 now, we can drop the arguments to super() ;)

artcz commented on conference/models.py:
```diff
@@ -200,7 +201,7 @@ class DeadlineContent(models.Model):
     Content of a deadline.
     """
-    deadline = models.ForeignKey(Deadline)
+    deadline = models.ForeignKey(Deadline, on_delete=models.CASCADE)
```
btw, I think we can drop the Deadline (as a model) – we no longer use it anywhere for new conferences. But will need to double check that.

artcz commented on conference/models.py:
```diff
@@ -415,7 +416,7 @@ class Presence(models.Model):
     """
```
same here, we should double check what we use Presence for.

artcz replied:
(same here means same as Deadline, maybe we can drop it)

artcz commented on conference/models.py:
```diff
@@ -437,8 +438,8 @@ class AttendeeLink(models.Model):
     """
```
Same here, check if we use AttendeeLink, and maybe drop it if not used.

artcz commented on conference/models.py:
```diff
@@ -1259,16 +1262,16 @@ def split(self, time):
-    event = models.ForeignKey(Event)
-    user = models.ForeignKey('auth.User')
+    event = models.ForeignKey(Event, on_delete=models.CASCADE)
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE)
```
will it still complain  if we used `User` here? (class, not a string)

umgelurgel replied:
Not sure, but I'd rather use either `get_user_model` or `settings.AUTH_USER_MODEL` so that when we do migrate to a custom `User` class, the foreign keys don't need to be updated.

artcz commented on conference/tests/test_comment_views.py:
```diff
@@ -0,0 +1,91 @@
+# coding: utf-8
```
is that integral part of the django20 warnings branch?

artcz commented on conference/utils.py:
```diff
@@ -114,23 +114,21 @@ def voting_results():
-                language = pieces[3].strip()
-                tid = int(pieces[1].strip())
-                results.append((tid, type, language))
+            with open(settings.TALKS_RANKING_FILE) as ranking_file:
```
:+1:

artcz reviewed (changes requested):
LGTM, i left some comments, and not sure if hcomments should be part of this diff or not.

umgelurgel committed:
Remove Django 2.0 warnings from the application.

Fix conference models django 20 depracation warnings.
Fix p3 models django 20 deprecation warnings.
Fix assopy models django 20 deprecation warnings.
Fix user.is_authenticated/is_anonymous deprecation warnings.
Replace assignment_tag with simple_tag.
Fix remaining deprecation warnings.
Fix middleware and jsonify deprecation warnings.

artcz commented on assopy/auth_backends.py:
```diff
@@ -40,7 +40,7 @@ class IdBackend(_AssopyBackend):
     """
-    def authenticate(self, uid=None):
+    def authenticate(self, request, uid=None):
         try:
```
this I belive will conflict on rebase with current dev/ep2019

umgelurgel committed:
Address review comments.

artcz reviewed (approved):
LGTM
